wakuv2/waku.go

```diff
@@ -1800,15 +1777,40 @@ func (w *Waku) StopDiscV5() error {
 }
 
 func (w *Waku) ConnectionChanged(state connection.State) {
-	if !state.Offline && !w.onlineChecker.IsOnline() {
-		select {
-		case w.goingOnline <- struct{}{}:
-		default:
-			w.logger.Warn("could not write on connection changed channel")
+	isOnline := !state.Offline
+	if w.cfg.LightClient {
+		//TODO: Update this as per  https://github.com/waku-org/go-waku/issues/1114
+		//Trigger FilterManager to take care of any pending filter subscriptions
+		go w.filterManager.onConnectionStatusChange(w.cfg.DefaultShardPubsubTopic, isOnline)
+
+		//If connection state is reported by something other than peerCount becoming 0 e.g from mobile app, disconnect all peers
+		if (state.Offline && len(w.node.Host().Network().Peers()) > 0) ||
+			(w.state.Type != state.Type && !w.state.Offline && !state.Offline) { // network switched between wifi and cellular
```

Comment:
How about extract `network switched between wifi and cellular` logic to its own method?